modelz/console.py

```diff
@@ -30,3 +32,12 @@ def print(self, msg: str):
 
 
 console = ModelzConsole() if config.disable_rich else Console()
+
+
+def jsonFormattedPrint(resp: Response):
+    try:
+        formatted = json.dumps(resp.parsed.to_dict(), indent=2)
+    except Exception:
+        formatted = resp.content.decode()
+    finally:
+        console.print(formatted)
```

Comment:
I'm quite confused here. Is this `Request` manually created or auto-generated? Why does the `parse` only work for HTTP 201? Why do you need to use the `resp.content.decode()` directly? Why is it a function instead of a method of the `Response`?

Reply:
> I'm quite confused here. Is this `Request` manually created or auto-generated? Why does the `parse` only work for HTTP 201? Why do you need to use the `resp.content.decode()` directly? Why is it a function instead of a method of the `Response`?

@kemingy The Response is generated and has a type definition:
```python
@define
class Response(Generic[T]):
    """A response from an endpoint"""

    status_code: HTTPStatus
    content: bytes
    headers: MutableMapping[str, str]
    parsed: Optional[T]
```
Sadly, it doesn't have any method.
Most `Response.parsed` like `DeploymentResponse` would has method `to_dict()`, but some of them might be `None` like `delete deployment`, so this catch is not designed for http error code, but different Response type.

Reply:
I give up. The generated code doesn't make sense and doesn't has to make sense.